sys/net/application_layer/gcoap/forward_proxy.c

```diff
@@ -84,8 +89,34 @@ gcoap_listener_t forward_proxy_listener = {
     _request_matcher_forward_proxy
 };
 
+static void *_forward_proxy_thread_start(void *arg)
+{
+    (void)arg;
+
+    msg_t msg;
+
+    while (1) {
+        msg_receive(&msg);
+        client_ep_t *cep = (client_ep_t *)msg.content.ptr;
+        switch (msg.type) {
+            case FORWARD_PROXY_MSG_SEND:
+                gcoap_req_send((uint8_t *)cep->pdu.hdr, coap_get_total_len(&cep->pdu),
+                                &cep->server_ep, _forward_resp_handler, cep);
+                break;
+            default:
+                DEBUG_PUTS("gcoap_forward_proxy: unknown message type\n");
+                break;
+        }
+    }
+    return NULL;
+}
+
 void gcoap_forward_proxy_init(void)
 {
+    _forward_proxy_pid = thread_create(_forward_proxy_thread, sizeof(_forward_proxy_thread),
+                                        THREAD_PRIORITY_MAIN - 1,
+                                        THREAD_CREATE_STACKTEST, _forward_proxy_thread_start,
+                                        NULL, "proxy");
```

Comment:
And

```suggestion
    if IS_USED(MODULE_GCOAP_FORWARD_PROXY_THREAD) {
        _forward_proxy_pid = thread_create(_forward_proxy_thread, sizeof(_forward_proxy_thread),
                                            THREAD_PRIORITY_MAIN - 1,
                                            THREAD_CREATE_STACKTEST, _forward_proxy_thread_start,
                                            NULL, "proxy");
    }
```

(probably needs to be adapted for the fact that `_forward_proxy_thread_start` will become at least an `extern`)